### DISCUSSION TITLE
ESP selected proposal is not reporting "modp2048" setting (DH group) in /var/log/charon.log

### DISCUSSION THREAD

**Discussion by marcogaravello** [strongswan/strongswan#994] [Q&A] [answered]
Hello,

I'm experiencing a discrepancy on _/var/log/charon.log_: when tunnel starts and after IKE_SA <tunnel_name> is established, following rows are logged

_Apr 15 06:26:08 14[IKE] <tunnel_name|20761> IKE_SA tunnel_name[20761] established between IP_peer_1[IP_peer_1]...IP_peer_2[IP_peer_2]
Apr 15 06:26:08 14[IKE] <tunnel_name|20761> scheduling reauthentication in 28063s
Apr 15 06:26:08 14[IKE] <tunnel_name|20761> maximum IKE_SA lifetime 28603s
**Apr 15 06:26:08 14[CFG] <tunnel_name|20761> selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/NO_EXT_SEQ**
Apr 15 06:26:08 14[IKE] <tunnel_name|20761> CHILD_SA tunnel_name{1193531} established with SPIs c5ec51a9_i c8268943_o and TS private_subnet_peer_1 === private_subnet_peer_2_

the bolded row is not reporting "modp2048" setting, while such setting is set in _/etc/ipsec.conf_ and working (both endpoints have such setting enabled).
To be clear, I was expecting bolded row to be like this

_Apr 15 06:26:08 14[CFG] <tunnel_name|20761> selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/**MODP2048**/NO_EXT_SEQ_

For this reason I get confused watching the log as it doesn't let you understand which are the real settings being used..
Is this some kind of bug?

Thanks in advance

**Comment by tobiasbrunner** [comment, marked as answer]
Please read [this](https://docs.strongswan.org/strongswan-docs/5.9/config/rekeying.html#_ikev2).

**Reply by marcogaravello** [reply]
Thank you so much!
